### PR TITLE
Revised "asynchronous programming with async and await (C#)" page

### PR DESCRIPTION
**Pull request by pkulikov**
Mainly, revised [Asynchronous programming with async and await (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/) page:
- Updated definition of an async method: only `async` keyword is necessary.
- Added link to UWP page on asynchronous programming.
- Removed link repetitions
- Added code formatting when necessary

In addition, cross-referenced various pages on async programming within the documentation.

### CODE REVIEW

File: docs/csharp/programming-guide/concepts/async/index.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ For more information about control flow, see [Control Flow in Async Programs (C#
...
   
- The Windows Runtime also contains many methods that you can use with `async` and `await` in Windows apps. For more information and example methods, see [Quickstart: using the await operator for asynchronous programming](/previous-versions/windows/apps/hh452713(v=win.10)), [Asynchronous programming (Windows Store apps)](/previous-versions/windows/apps/hh464924(v=win.10)), and [WhenAny: Bridging between the .NET Framework and the Windows Runtime](https://msdn.microsoft.com/library/jj635140(v=vs.120).aspx).  
+ The Windows Runtime also contains many methods that you can use with `async` and `await` in Windows apps. For more information, see [Threading and async programming](/windows/uwp/threading-async/) for UWP development and [Asynchronous programming (Windows Store apps)](/previous-versions/windows/apps/hh464924(v=win.10)), [Quickstart: using the await operator for asynchronous programming](/previous-versions/windows/apps/hh452713(v=win.10)) if you use earlier versions of the Windows Runtime.
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/concepts/async/index.md]: Although this is unrelated to your PR:
nit: comma after "for UWP development"
nit: no comma after the Asychronous programming (Windows Store apps) link, and add an "and" between that link and the Quckstart link.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thank you, @pkulikov, for making yet another set of improvements in the documentation. There are just two minor comments unrelated to your PR; I'd appreciate it if you could make the corrections.

File: docs/csharp/programming-guide/concepts/async/index.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/programming-guide/concepts/async/index.md]: @rpetrusha thank you for feedback. I've pushed changes: all commas are ordered now :).

File: docs/csharp/programming-guide/concepts/async/index.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ Asynchrony proves especially valuable for applications that access the UI thread
...
   
 ##  <a name="BKMK_HowtoWriteanAsyncMethod"></a> Async methods are easier to write  
- The [async](../../../../csharp/language-reference/keywords/async.md) and [await](../../../../csharp/language-reference/keywords/await.md) keywords in C# are the heart of async programming. By using those two keywords, you can use resources in the .NET Framework, .NET Core, or the Windows Runtime to create an asynchronous method almost as easily as you create a synchronous method. Asynchronous methods that you define by using `async` and `await` are referred to as *async methods*.  
+ The [async](../../../../csharp/language-reference/keywords/async.md) and [await](../../../../csharp/language-reference/keywords/await.md) keywords in C# are the heart of async programming. By using those two keywords, you can use resources in the .NET Framework, .NET Core, or the Windows Runtime to create an asynchronous method almost as easily as you create a synchronous method. Asynchronous methods that you define by using `async` keyword are referred to as *async methods*.
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/concepts/async/index.md]: I apologize, @pkulikov -- I missed this one earlier. There should be a "the" before `async` in the last sentence.

Comment (pkulikov) [reply docs/csharp/programming-guide/concepts/async/index.md]: @rpetrusha that's fixed now.

**Review by rpetrusha — APPROVED** [review]
Thanks, @pkulikov, for making the additional correction, and again, I apologize for having missed it in the first review. We'll merge your PR now.
